clock-bound-d/src/server.rs

```diff
@@ -56,7 +57,7 @@ impl ClockBoundServer {
     ) -> Result<(), io::Error> {
         let mut request: [u8; 12] = [0; 12];
 
-        let (request_size, client) = match self.socket.recv_from(&mut request) {
+        let (request_size, client) = match self.socket.recv_from_unix_addr(&mut request) {
```

Comment:
You can use `?` notation to do this more idomatically, as simply `let (request_size_client) = self.socket.recv_from_unix_addr(&mut request)?;`